src/test/testing_assert.cpp

```diff
@@ -41,47 +43,54 @@ Matrix _table_to_matrix(const opossum::Table& t) {
   return matrix;
 }
 
-void _print_matrix(const Matrix& m) {
-  std::cout << "-------------" << std::endl;
+std::string _matrix_to_string(const std::string& title, const Matrix& m) {
+  std::stringstream stream;
+  stream << "-------" << title << "-------" << std::endl;
   for (unsigned row = 0; row < m.size(); row++) {
     for (opossum::ColumnID col{0}; col < m[row].size(); col++) {
-      std::cout << std::setw(8) << m[row][col] << " ";
+      stream << std::setw(8) << m[row][col] << " ";
     }
-    std::cout << std::endl;
+    stream << std::endl;
   }
-  std::cout << "-------------" << std::endl;
+  stream << "---------------------------" << std::endl;
+  return stream.str();
 }
 }  // namespace
 
 namespace opossum {
 
-::testing::AssertionResult check_table_equal(const Table& tleft, const Table& tright, bool order_sensitive,
-                                             bool strict_types) {
-  Matrix left = _table_to_matrix(tleft);
-  Matrix right = _table_to_matrix(tright);
-
-  const auto print_tables = [&]() {
-    std::cout << "== Tables are not equal ==" << std::endl;
-    _print_matrix(left);
-    _print_matrix(right);
-    std::cout << "==========================" << std::endl;
+bool check_table_equal(const std::shared_ptr<const Table>& opossum_table,
+                       const std::shared_ptr<const Table>& expected_table, OrderSensitivity order_sensitivity,
```

Comment:
Who says that the expected table is not an opossum table?

Reply:
Example, pls. In my mind there is always one table generated by calling opossum and one that serves as a reference to determine whether opossum was right.

Reply:
And it REALLY helps if there is debug output that clearly states which is the "wrong" table (i.e. opossum's) and how the table should actually look like

Reply:
```
delete_test.cpp:130
EXPECT_TABLE_EQ(validate->get_output(), expected_result->get_output());
```

Both sides are tables within Hyrise. This is why I like expected/actual (as used by googletest) better.

Reply:
alrite, changed it